server/src/main/java/org/elasticsearch/health/SimpleHealthIndicatorDetails.java

```diff
@@ -24,4 +24,21 @@ public record SimpleHealthIndicatorDetails(Map<String, Object> details) implemen
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         return builder.map(details);
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
I think `equals&hashCode` are redundant here since `SimpleHealthIndicatorDetails` is a record

Reply:
Good catch, this did not merged properly